test/integration/dataset_collection.py

```diff
@@ -12,41 +14,76 @@ class IntegrationDatasetCollection(TestCase):
 
     def test_dataset_collection(self):
         samples_path = Path(__file__).absolute().parent / "dataset_collection"
-        results_predefined = [
-            ['GlobalShortcutConfigForm.java',
-             'GlobalShortcutConfigForm', 'loadConfig();', 180, 186, 'initComponents'],
-            ['GlobalShortcutConfigForm.java',
-             'GlobalShortcutConfigForm', 'refresh();', 238, 318, 'loadConfig'],
-            ['GlobalShortcutConfigForm.java',
-             'GlobalShortcutConfigForm', 'this.initComponents();', 82, 88, 'GlobalShortcutConfigForm'],
-            ['HudFragment.java', 'HudFragment', 'toggleMenus();',
-             131, 510, 'build'],
-            ['HudFragment.java', 'HudFragment',
-             'showLaunchConfirm();', 590, 479, 'addWaveTable'],
-            ['PlanetDialog.java', 'PlanetDialog', 'makeBloom();', 70,
-             147, 'PlanetDialog'],
-            ['PlanetDialog.java', 'PlanetDialog',
-             'updateSelected();', 136, 334, 'PlanetDialog'],
-            ['ReaderHandler.java', 'ReaderHandler',
-             'receiveMessage();', 183, 115, 'onWebSocketConnect'],
-            ['ReaderHandler.java', 'ReaderHandler',
-             'receiveMessage();', 196, 115, 'onWebSocketText'],
-            ['ReaderHandler.java', 'ReaderHandler',
-             'final int receiverQueueSize = getReceiverQueueSize();', 76, 253, 'ReaderHandler'],
-            ['ToggleProfilingPointAction.java',
-             'ToggleProfilingPointAction', 'ProfilingPointsSwitcher chooserFrame = getChooserFrame();', 272, 311,
-             'actionPerformed'],
-            ['ToggleProfilingPointAction.java',
-             'ToggleProfilingPointAction', 'nextFactory();', 284, 361, 'actionPerformed'],
-            ['ToggleProfilingPointAction.java',
-             'ToggleProfilingPointAction', 'resetFactories();', 289, 369, 'actionPerformed'],
-            ['ToggleProfilingPointAction.java',
-             'ToggleProfilingPointAction',
-             'if (acceleratorModifiers != eventKeyStroke.getModifiers()) modifierKeyStateChanged();', 308, 324,
-             'eventDispatched'],
-            ['ToggleProfilingPointAction.java',
-             'ToggleProfilingPointAction', 'ProfilingPointsSwitcher chooserFrame = getChooserFrame();', 329, 311,
-             'modifierKeyStateChanged']]
+        # ignore output filename, cause it is not so important
+        results_predefined: List[Dict[str, any]] = \
+            [{'input_filename': 'GlobalShortcutConfigForm.java',
+              'class_name': 'GlobalShortcutConfigForm',
+              'invocation_text_string': 'this.initComponents();',
+              'method_where_invocation_occurred': 'GlobalShortcutConfigForm',
+              'start_line_of_function_where_invocation_occurred': 78,
```

Comment:
could you make it look more pretty?

Reply:
it's test .What do u mean pretty? It passed all flake8 rules. Actually, it has similar structure, but now we have a dict instead of list of lists